packages/cli/src/lib/implementation/core-config.middleware.integration.test.ts

```diff
@@ -1,28 +1,49 @@
-import { dirname, join } from 'node:path';
-import { fileURLToPath } from 'node:url';
-import { describe, expect } from 'vitest';
+import { describe, expect, vi } from 'vitest';
+import { ConfigPathError } from '@code-pushup/core';
+import { CoreConfig } from '@code-pushup/models';
+import { CORE_CONFIG_MOCK } from '@code-pushup/testing-utils';
 import { coreConfigMiddleware } from './core-config.middleware';
 
-const configDirPath = join(
-  fileURLToPath(dirname(import.meta.url)),
-  '..',
-  '..',
-  '..',
-  '..',
-  '..',
-  'testing-utils',
-  'src',
-  'lib',
-  'fixtures',
-  'configs',
-);
+vi.mock('@code-pushup/core', async () => {
+  const core = await vi.importActual('@code-pushup/core');
+  return {
+    ...(core as object),
```

Comment:
You can use `typeof import('...')` with `vi.importActual('...')` to get the accurate types instead of `uknown`:

```suggestion
  const core: typeof import('@code-pushup/core') = await vi.importActual('@code-pushup/core');
  return {
    ...core,
```

Reply:
There's a problem with the Nx lint rule: https://github.com/code-pushup/cli/actions/runs/7658472674/job/20873664503?pr=361

We encountered it in #431 as well. So in the end using `const core: object` is probably the best possible way.

Reply:
reverted to object